client/src/parent/chores/ChoreCard.tsx

```diff
@@ -126,77 +125,137 @@ export function ChoreCard(props: Prop): JSX.Element {
   }
 
-  function getDescription() {
-    if (!props.chore.description) return;
-
-    return (
-      <Typography
-        className={classes.description}
-        variant="body2"
-        color="textSecondary"
-        component="p"
-      >
-        {props.chore.description}
-      </Typography>
+  function handleDescriptionEdit(description: string) {
+    console.log("Editing description...");
+    var updatedChore = produce(
+      props.chore,
+      (draftChore: ChoreWithAssignments) => {
+        draftChore.description = description;
+      }
     );
+
+    putChoreMutation.mutate({
+      url: "/api/Chores/",
+      object: updatedChore,
+    });
+    console.log("Description edited.");
+    showInfo("Description edited."); // TODO js (11.03.2021): Replace dummy implementation.
   }
 
   return (
     <Card elevation={5}>
       <BiAvatarCardHeader
-        leftAvatarLabel={props.chore.assignments.length.toString()}
-        leftAvatarNotifications={
-          props.chore.assignments.filter(
-            (assignment) =>
-              assignment.state === AssignmentState.RequestedToCheck
-          ).length
+        leftAvatarComponent={
+          <Badge
+            badgeContent={
+              props.chore.assignments.filter(
+                (assignment) =>
+                  assignment.state === AssignmentState.RequestedToCheck
+              ).length
+            }
+            color="secondary"
+          >
+            <Avatar className={classes.avatar} onClick={handleExpandClick}>
+              {props.chore.assignments.length.toString()}
+            </Avatar>
+          </Badge>
+        }
+        titleComponent={
+          <EditableText
+            text={props.chore.name}
+            editable={!cardLocked}
+            editLabel="Bezeichnung des Ämtlis"
+            validationSchema={Yup.object({
+              text: Yup.string()
+                .required("Bitte definieren")
+                .max(50, "Maximum 50 Zeichen"),
+            })}
+            onTextChanged={handleTitleEdit}
+          />
         }
-        onLeftAvatarClick={handleExpandClick}
-        title={props.chore.name}
-        subtitle={new Date(props.chore.dueDate).toLocaleDateString("de-DE", {
-          weekday: "short",
-          year: "2-digit",
-          month: "short",
-          day: "numeric",
-        })}
-        rightAvatarLabel={props.chore.value.toString()}
-        rightAvatarNotifications={
-          props.chore.assignments.filter(
-            (assignment) =>
-              assignment.state === AssignmentState.CheckConfirmed ||
-              assignment.state === AssignmentState.Archived
-          ).length
+        subtitleComponent={
+          <EditableDate
+            date={props.chore.dueDate}
+            editable={!cardLocked}
+            editLabel="Ablaufdatum"
+            validationSchema={Yup.object({
+              date: Yup.date().min(
```

Comment:
@marZeu8 Hier die Validierung eines Datums mit Yup.
Die ValidationSchemas für die verschiedenen Felder könnten wir vielleicht irgendwo zentral ablegen, damit wir im Add-Dialog und beim Editieren die gleichen Regeln anwenden können. Das könnten wir vielleicht in einem späteren Schritt noch refactoren.